portal_welcome_email_template/wizards/portal_wizard.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Incaser Informatica S.L. - Sergio Teruel
+# (c) 2015 Incaser Informatica S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api
+from openerp import exceptions
+from openerp.tools.translate import _
+
+
+class WizardUser(models.TransientModel):
+    """
+        A model to configure users in the portal wizard.
+    """
+    _inherit = 'portal.wizard.user'
+
+    @api.model
+    def _send_email(self, wizard_user):
+        """ send notification email to a new portal user
+            @param wizard_user: browse record of model portal.wizard.user
+            @return: the id of the created mail.mail record
+        """
+        this_context = self._context
+        this_user = self.env['res.users'].sudo().browse(self._uid)
+        if not this_user.email:
+            raise exceptions.Warning(
+                _('Email Required'),
+                _('You must have an email address in your User Preferences to '
+                  'send emails.'))
+
+        # determine subject and body in the portal user's language
+        user = self.sudo()._retrieve_user(wizard_user)
+        context = dict(this_context or {}, lang=user.lang)
+        ctx_portal_url = dict(context, signup_force_type_in_url='')
+        portal_url = user.partner_id.with_context(
```

Comment:
you don't need to replace the whole context, `.with_context(lang=user.lang, signup_force_type_in_url='')` would suffice here